File: src/main/java/com/social/meow/repository/PostRepository.java
package com.social.meow.repository;

import com.social.meow.model.Post;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

@Repository
public interface  PostRepository extends JpaRepository<Post, Long> {
    @Query("SELECT p FROM Post p WHERE p.id < :lastId ORDER BY p.id DESC")
    Page<Post> findPostsBeforeLastId(Long lastId, Pageable pageable);
}

File: src/main/kotlin/com/social/meowkotlin/repository/PostRepository.kt
package com.social.meowkotlin.repository

import com.social.meowkotlin.model.Post
import org.springframework.data.jpa.repository.JpaRepository

@org.springframework.stereotype.Repository
interface PostRepository : JpaRepository<Post, Long> {
    fun findByIdLessThanOrderByIdDesc(lastId: Long, pageable: org.springframework.data.domain.Pageable): List<Post>
}
